src/lib/cache.js

```diff
@@ -110,25 +125,45 @@ function _set(key, data) {
   }
   /* eslint-enable no-param-reassign */
 
-  return deflateP(data).then(deflatedData => {
-    const payload = deflatedData.toString('base64')
-    verbose(`CACHE SET: ${key}: ${payload}`)
-    return client.set(
-      key,
-      payload,
-      CACHE_LIFETIME_IN_SECONDS,
-      err => {
-        if (err) error(err)
-      }
-    )
-  }).catch(err => {
-    error(err)
-  })
+  if (CACHE_COMPRESSION_DISABLED) {
+    return new Promise((resolve, reject) => {
+      const payload = JSON.stringify(data)
+      verbose(`CACHE SET: ${cacheKey(key)}: ${payload}`)
+      resolve(
+        client.set(
+          cacheKey(key),
+          payload,
+          CACHE_LIFETIME_IN_SECONDS,
+          err => {
+            if (err) error(err)
+          }
+        )
+      )
+    }).catch(err => {
+      error(err)
+    })
+  } else {
+    return deflateP(data).then(deflatedData => {
+      const payload = deflatedData.toString('base64')
+      verbose(`CACHE SET: ${cacheKey(key)}: ${payload}`)
+
+      return client.set(
+        cacheKey(key),
+        payload,
+        CACHE_LIFETIME_IN_SECONDS,
+        err => {
+          if (err) error(err)
+        }
+      )
+    }).catch(err => {
+      error(err)
+    })
```

Comment:
FYI Because the `error` function takes a single `err` parameter, you don’t even really need to create this extra closure and could just do `.catch(error)`.

Reply:
Righto